lapce-app/src/file_explorer/view.rs

```diff
@@ -1,30 +1,67 @@
-use std::rc::Rc;
+use std::{path::Path, rc::Rc};
 
 use floem::{
     cosmic_text::Style as FontStyle,
     event::{Event, EventListener},
     peniko::Color,
     reactive::{create_rw_signal, RwSignal},
-    style::{CursorStyle, Style},
+    style::{AlignItems, CursorStyle, Position, Style},
     view::View,
     views::{
-        container, label, list, scroll, stack, svg, virtual_list, Decorators,
-        VirtualListDirection, VirtualListItemSize,
+        container, container_box, label, list, scroll, stack, svg, virtual_list,
+        Decorators, VirtualListDirection, VirtualListItemSize,
     },
     EventPropagation,
 };
+use lapce_core::selection::Selection;
+use lapce_rpc::file::{FileNodeViewData, IsRenaming, RenameState};
+use lapce_xi_rope::Rope;
 
 use super::{data::FileExplorerData, node::FileNodeVirtualList};
 use crate::{
     app::clickable_icon,
     command::InternalCommand,
     config::{color::LapceColor, icon::LapceIcons},
     editor_tab::{EditorTabChild, EditorTabData},
-    panel::{position::PanelPosition, view::panel_header},
+    panel::{kind::PanelKind, position::PanelPosition, view::panel_header},
     plugin::PluginData,
-    window_tab::WindowTabData,
+    text_input::text_input,
+    window_tab::{Focus, WindowTabData},
 };
 
+/// Blends `foreground` with `background`.
+///
+/// Uses the alpha channel from `foreground` - if `foreground` is opaque, `foreground` will be
+/// returned unchanged.
+///
+/// The result is always opaque regardless of the transparency of the inputs.
+fn blend_colors(background: Color, foreground: Color) -> Color {
```

Comment:
I don't like having this function here. The problem is that `ERROR_LENS_ERROR_BACKGROUND` is transparent, so other filenames would be visible through the error box if it were used for the error box background. Just setting the alpha channel to 255 would result in the colour looking wrong.

Perhaps a new colour should be defined for the error box background, or maybe an existing dependency has a blend function that I didn't spot?